cloudinit/cmd/devel/logs.py

```diff
@@ -143,15 +144,31 @@ def _copytree_rundir_ignore_files(curdir, files):
 
 def _write_command_output_to_file(cmd, filename, msg, verbosity):
     """Helper which runs a command and writes output or error to filename."""
+    output = ""
     try:
-        out, _ = subp(cmd)
+        ensure_dir(os.path.dirname(filename))
+        output = subp(cmd)[0]
```

Comment:
This is the only line that can raise a `ProcessExecutionError`, so we shouldn't need anything else in the try

Reply:
To clarify, do you mean that the ensure_dir command (line 149) should go before (outside) the try block?